cli/pkg/local/app.go

```diff
@@ -82,18 +84,58 @@ func NewApp(ctx context.Context, ver config.Version, verbose, strict, reset bool
 		return nil, err
 	}
 
+	// Get full path to project
+	projectPath, err = filepath.Abs(projectPath)
+	if err != nil {
+		return nil, err
+	}
+	dbDirPath := filepath.Join(projectPath, DefaultDBDir)
+	err = os.MkdirAll(dbDirPath, os.ModePerm) // Create project dir and db dir if it doesn't exist
+	if err != nil {
+		return nil, err
+	}
+
+	// backward compatibility for old stage.db file
+	embedCatalog := false
+	_, err = os.Stat(filepath.Join(projectPath, "stage.db"))
+	if err == nil { // a old stage.db file exists
+		// rename to main.db and move to db directory
+		err := os.Rename(filepath.Join(projectPath, "stage.db"), filepath.Join(dbDirPath, DefaultOLAPDSN))
+		if err != nil {
+			return nil, err
+		}
+
+		err = os.Rename(filepath.Join(projectPath, "stage.db.wal"), filepath.Join(dbDirPath, fmt.Sprintf("%s.wal", DefaultOLAPDSN)))
+		if err != nil {
+			return nil, err
+		}
+		// catalog is embedded for older flow
+		embedCatalog = true
```

Comment:
This doesn't really work – on the next restart, `embedCatalog` will be false (since `stage.db` was moved), and it will rebuild everything. Should either not move `stage.db` and use it directly (probably best) or just drop it and rebuild.

Reply:
This seems to be getting somewhat complicated now. How about we keep it clean and just drop and rebuild which may anyways be required if we merge this in a release with duckdb upgrade ?

Reply:
Maybe I'm missing something, seems quite simple to say "if `stage.db` exists, set DSN to it and `embedCatalog = true`? But if I'm missing something or it gets ugly, sure let's just drop and rebuild

Reply:
Yeah I mean we will always have to carry these if else conditions. Data will be split between tmp directory (catalog) and stage.db. 
In this case it seems better to just drop and rebuild which we anyways do from time to time.